src/Microsoft.PowerShell.Commands.Utility/commands/utility/TestJsonCommand.cs

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using System.Management.Automation.Internal;
+
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using NJsonSchema;
+
+namespace Microsoft.PowerShell.Commands
+{
+    /// <summary>
+    /// This class implements Test-Json command.
+    /// </summary>
+    [Cmdlet(VerbsDiagnostic.Test, "Json", HelpUri = "")]
+    public class TestJsonCommand : PSCmdlet
+    {
+        /// <summary>
+        /// An Json to be validated.
+        /// </summary>
+        [Parameter(Position = 0, Mandatory = true, ValueFromPipeline = true)]
+        public String Json { get; set; }
+
+        /// <summary>
+        /// A schema to validate the Json against.
+        /// It is optional parameter.
+        /// If the parameter is absent the cmdlet only try to parse the Json.
+        /// If the parameter present the cmdlet try to parse the Json and
+        /// then check the Json against the schema. Before the check
+        /// the cmdlet parse the schema doing implicitly check the schema too.
+        /// </summary>
+        [Parameter(Position = 1)]
+        [ValidateNotNullOrEmpty()]
+        public String Schema { get; set; }
+
+        private JsonSchema4 _jschema;
+
+        /// <summary>
+        /// Prepare an Json schema.
+        /// </summary>
+        protected override void BeginProcessing()
+        {
+            if (Schema != null)
+            {
+                try
+                {
+                    _jschema = JsonSchema4.FromJsonAsync(Schema).Result;
+                }
+                catch (Exception exc)
+                {
+                    Exception exception = new Exception(TestJsonCmdletStrings.InvalidJsonSchema, exc);
+                    ThrowTerminatingError(new ErrorRecord(exception, "InvalidJsonSchema", ErrorCategory.InvalidData, null));
+                }
+            }
+        }
+
+        /// <summary>
+        /// Validate an Json.
+        /// </summary>
+        protected override void ProcessRecord()
+        {
+            JObject parsedJson = null;
+            bool result = true;
+
+            try
+            {
+                parsedJson = JObject.Parse(Json);
+
+                if (_jschema != null)
+                {
+                    var errorMessages = _jschema.Validate(parsedJson);
+                    if (errorMessages.Count != 0)
```

Comment:
You're tests for errorMessages are a little confusing.  You're checking count here but then checking for null a few lines further on.   If errorMessages can be null, shouldn't this line should read something like...

if (null != errorMessages && errorMessages.Count != 0)

Reply:
Fixed.

Reply:
Closed

Reply:
@iSazonov: I don't see a change.

Reply:
It is in "Fix null check" commit.

Reply:
Closed